src/OpenTelemetry.Instrumentation.EntityFrameworkCore/Implementation/EntityFrameworkDiagnosticListener.cs

```diff
@@ -139,10 +142,19 @@ public override void OnEventWritten(string name, object? payload)
                         }
 
                         var dataSource = (string)this.dataSourceFetcher.Fetch(connection);
-                        activity.AddTag(AttributeDbName, database);
                         if (!string.IsNullOrEmpty(dataSource))
                         {
-                            activity.AddTag(AttributePeerService, dataSource);
+                            activity.AddTag(AttributeServerAddress, dataSource);
+                        }
+
+                        if (this.options.EmitOldAttributes)
+                        {
+                            activity.AddTag(AttributeDbName, database);
+                        }
+
+                        if (this.options.EmitNewAttributes)
+                        {
+                            activity.AddTag(AttributeDbNamespace, database);
```

Comment:
@trask @lmolkova what are your thoughts here?

Per [footnote 4 in the conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/database/database-spans.md#common-attributes):

> Semantic conventions for individual database systems SHOULD document what db.namespace means in the context of that system.

Since this instrumentation is for an ORM and supports many database systems, this should respect the conventions declared for the target database system.

That said, not all systems this instrumentation supports have documentation for what `db.namespace` means.

Reply:
we have a similar issue in Java (https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/jdbc/library/src/test/java/io/opentelemetry/instrumentation/jdbc/internal/JdbcConnectionUrlParserTest.java)

technically, I think stable JDBC instrumentation should only emit stable instrumentation by default, which means only emitting instrumentation for stable `db.system` values

in practice, this would mean that we can't mark the JDBC instrumentation as stable even after we update it to emit the stable database semconv (unless we want to force lots of users to opt-in to the unstable `db.system` values, which I don't think is a great experience)

I'm hoping this will motivate us to push semconv definitions for `db.namespace` for all of these other databases quickly, we just haven't had the bandwidth to do it yet and didn't want to block initial database semconv stability on it